src/test/scala/dev/kamu/cli/MetadataRepositorySpec.scala

```diff
@@ -53,4 +58,56 @@ class MetadataRepositorySpec extends FunSuite with Matchers with KamuTestBase {
       actual3 shouldEqual Seq(dsC.id, dsB.id, dsA.id)
     }
   }
+
+  test(raw"'kamu add' from HTTP") {
+    withEmptyRepo { kamu =>
+      val testDir = getRandomDir()
+      val serverPort = 8080
+      val hostPort = 9000
+      val path: Path = new Path(testDir, raw"test-dataset.yaml")
+      fileSystem.mkdirs(testDir)
+      val testServerName = s"kamu_test_http_server"
+      val testHttpServerArgs = DockerRunArgs(
+        image = "python:3.7-slim",
+        args = List("python3", "-m", "http.server", s"${serverPort}"),
+        exposePortMap = Map(hostPort -> serverPort),
+        volumeMap = Map(testDir -> new Path("/mnt/kamu")),
+        containerName = Some(testServerName),
+        detached = true
+      )
+      val testHttpServer = new DockerClient(FileSystem.get(new Configuration()))
```

Comment:
`KamuTestBase` should already have `fileSystem` member, so you don't need to create it anew